### PR TITLE
Accelerated One electron Coulomb integrals

### PR DESCRIPTION
**Pull request by henryw7**
Not done, TODO list:

- [x] Add tests
- [ ] Performance benchmark against int2e with fake mol
- [x] Split grid points into batches
- [ ] More angular momentum integral special cases
- [ ] Change density order and merge with int2e J engine
- [ ] Redefine variables according to new conventions

### CODE REVIEW

File: gpu4pyscf/lib/gint/g3c1e.cu [hunk 1]
```diff
@@ -0,0 +1,124 @@
+#include "gint.h"
```
Comment (wxj6000) [inline gpu4pyscf/lib/gint/g3c1e.cu:1]: License

File: gpu4pyscf/lib/gint/g1e_root_123.cu [hunk 2]
```diff
@@ -0,0 +1,312 @@
+/*
+ * gpu4pyscf is a plugin to use Nvidia GPU in PySCF package
+ *
```
Comment (wxj6000) [inline gpu4pyscf/lib/gint/g1e_root_123.cu:3]: Recent format

File: gpu4pyscf/gto/moleintor.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+from gpu4pyscf.scf.int4c2e import BasisProdCache
+from gpu4pyscf.df.int3c2e import sort_mol, _split_l_ctr_groups, get_pairing
+from gpu4pyscf.gto.mole import basis_seg_contraction
+
+BLKSIZE = 128
+
+libgvhf = load_library('libgvhf')
+libgint = load_library('libgint')
+
+class VHFOpt(_vhf.VHFOpt):
```
Comment (wxj6000) [inline gpu4pyscf/gto/moleintor.py:32]: This function has been defined in int3c2e.py. No need to redefine here.

File: gpu4pyscf/gto/moleintor.py [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+                int3c_angular_slice = cart2sph(int3c_angular_slice, axis=1, ang=lj)
+                int3c_angular_slice = cart2sph(int3c_angular_slice, axis=2, ang=li)
+            int3c_grid_slice[:, j0:j1, i0:i1] = int3c_angular_slice
+        row, col = np.tril_indices(nao)
+        int3c_grid_slice[:, row, col] = int3c_grid_slice[:, col, row]
+        ao_idx = np.argsort(intopt.ao_idx)
+        grid_idx = np.arange(ngrids_of_split)
+        int3c_grid_slice = int3c_grid_slice[np.ix_(grid_idx, ao_idx, ao_idx)]
+
+        cp.cuda.runtime.memcpy(int3c[i_grid_split : i_grid_split + ngrids_of_split, :, :].ctypes.data, int3c_grid_slice.data.ptr, int3c_grid_slice.nbytes, cp.cuda.runtime.memcpyDeviceToHost)
```
Comment (wxj6000) [inline gpu4pyscf/gto/moleintor.py:272]: .get()

File: gpu4pyscf/lib/gint/bpcache.cu [hunk 5]
```diff
@@ -53,6 +53,7 @@ void GINTdel_basis_prod(BasisProdCache **pbp)
     
     if (bpcache->aexyz != NULL) {
         free(bpcache->aexyz);
+        free(bpcache->h_bas_coords);
```
Comment (wxj6000) [inline gpu4pyscf/lib/gint/bpcache.cu:56]: bas_coords on host can be passed into CPU kernel directly. No need to save it in bpcache.

File: gpu4pyscf/lib/gint/g1e.cu [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+
+    if (j_l > 0) {
+        const double Bx = bas_x[jsh];
+        const double By = bas_y[jsh];
+        const double Bz = bas_z[jsh];
+        const double ABx = Ax - Bx;
+        const double ABy = Ay - By;
+        const double ABz = Az - Bz;
+
+        for (int i_root = 0; i_root < NROOTS; i_root++) {
```
Comment (wxj6000) [inline gpu4pyscf/lib/gint/g1e.cu:121]: Put `i_root` into inner loop.

**Review by sunqm — APPROVED** [review]

**Review by sunqm — APPROVED** [review]

**Pull request merged**